app/models/commit_status.rb

```diff
@@ -39,38 +46,77 @@ def expire_cache(commit)
 
   def combined_status
     @combined_status ||= begin
-      statuses = [github_status]
+      statuses = [github_state]
       statuses += [release_status, *ref_statuses].compact if @stage
-      statuses[1..-1].each_with_object(statuses[0]) { |status, merged| merge(merged, status) }
+      merge_statuses(statuses)
     end
   end
 
-  def merge(a, b)
-    a[:state] = [a.fetch(:state), b.fetch(:state)].max_by { |state| STATE_PRIORITY.index(state.to_sym) }
-    a.fetch(:statuses).concat b.fetch(:statuses)
-  end
-
+  # Gets a reference's state, combining results from both the Status and Checks API
   # NOTE: reply is an api object that does not support .fetch
-  def github_status
+  def github_state
     static = @reference.match?(Build::SHA1_REGEX) || @reference.match?(Release::VERSION_REGEX)
     expires_in = ->(reply) { cache_duration(reply) }
     cache_fetch_if static, cache_key(@reference), expires_in: expires_in do
-      GITHUB.combined_status(@project.repository_path, @reference).to_h
+      checks_result = with_octokit_client_error_rescue { github_check }
+      status_result = with_octokit_client_error_rescue { github_status }
+
+      results_with_statuses = [checks_result, status_result].select { |result| result[:statuses].any? }
```

Comment:
could do the merge first and then check empty after ... but either way is fine 🤷‍♂️